feat: Adds related dashboards tab to the Table detail page

Golodhros opened the pull request:
### Summary of Changes
Adds a related dashboards tab on the Table detail page

### Tests
Added some basic tests to table detail
Added tests to the helper functions

![base_rides_-_Amundsen_Table_Details](https://user-images.githubusercontent.com/190833/83467828-e162fb00-a42f-11ea-82fc-0fee2c8be50e.png)

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes. 
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

Golodhros commented on amundsen_application/static/js/ducks/tableMetadata/api/v0.ts:
```diff
@@ -14,26 +15,45 @@ export type TableData = TableMetadata & {
-      statusCode: response.status,
-    };
-  });
+export function getTableData(
+  tableKey: string,
+  index?: string,
+  source?: string
+) {
+  const relatedDashboardsSlug: string = getRelatedDashboardSlug(tableKey);
+  const relatedDashboardsURL: string = `${API_PATH}/table/${relatedDashboardsSlug}/dashboards`;
+  const relatedDashboardsRequest = axios.get<RelatedDashboardDataAPI>(
+    relatedDashboardsURL
+  );
+
+  const tableQueryParams = getTableQueryParams(tableKey, index, source);
+  const tableURL = `${API_PATH}/table?${tableQueryParams}`;
+  const tableRequest = axios.get<TableDataAPI>(tableURL);
+
+  return Promise.all([tableRequest, relatedDashboardsRequest]).then(
+    ([tableResponse, relatedDashboardsResponse]: [
+      TableDataAPI,
+      RelatedDashboardDataAPI
+    ]) => ({
+      data: getTableDataFromResponseData(
+        tableResponse.data,
+        relatedDashboardsResponse.data
+      ),
+      owners: getTableOwnersFromResponseData(tableResponse.data),
+      tags: tableResponse.data.tableData.tags,
+      statusCode: tableResponse.status
+    })
+  );
 }
```
This is the tricky part I can't get to type

Golodhros replied:
@dkempner

ttannis commented on amundsen_application/static/js/ducks/tableMetadata/api/helpers.ts:
```diff
@@ -13,10 +13,22 @@ export function getTableQueryParams(key: string, index?: string, source?: string
+export function getTableDataFromResponseData(responseData: API.TableDataAPI, relatedDashboardsData: API.RelatedDashboardDataAPI ): TableMetadata {
+  const mergedTableData = {
+    ...filterFromObj(responseData.tableData, ['owners', 'tags']),
+    ...relatedDashboardsData
```
Even though `RelatedDashboardDataAPI` is typed as  `{ dashboards: DashboardResource[]; }`, I think extra information like `msg` will still exist on that object and then get passed around and eventually set on the the returned object. Could you double check whether or not `mergedTableData` end up with only `dashboards` and not `msg`?

Golodhros replied:
I will double check!

Golodhros replied:
You are right! There is a msg and status_code that shouldn't be there, I will remove those.

Thanks!

Golodhros commented:
> Minor comment on visuals, and feel free to merge. Now that lists are in a `TabsComponent` where the tab headers have a bottom border, let's apply something like
> 
> ```
>   .right-panel {
>     .tabs-component {
>       .tab-content .list-group-item:first-child {
>         border-top: none;
>       }
>     }
>   }
> ```
> 
> in `components/TableDetail/styles.scss` to avoid the double-thick border under the tab headers.

Good catch!

I updated the styles and removed the specific one for the dashboard detail page.
